app/src/main/java/ch/epfl/favo/view/tabs/MapPage.java

```diff
@@ -105,14 +116,60 @@ public void onMapReady(GoogleMap googleMap) {
     mMap.setMyLocationEnabled(true);
     mMap.setInfoWindowAdapter(this);
     mMap.setOnInfoWindowClickListener(this);
-    try { // TODO: might want to move this to onCreateView (not sure if listeners are duplicated)
+    mMap.getUiSettings().setZoomControlsEnabled(true);
+    mMap.setPadding(0, 0, 0, 140);
+    mLocation = DependencyFactory.getCurrentGpsTracker(getContext()).getLocation();
+    mMap.setOnMapLongClickListener(new LongClick());
+    mMap.setOnMarkerDragListener(new MarkerDrag());
+    try { // might want to move this to onCreateView (not sure if listeners are duplicated)
+      // we should not move, because only after the map is ready, can we draw markers then
```

Comment:
The listeners should only be called from onCreateView. Since this method is called there it's no problem :)

Reply:
The listeners call drawing markers, I thought drawing markers too early will crash app because map is possibly not ready yet(mMap variable is null before map is ready)